scripts/wallet-operator-mapper/query-proof-of-funds.js

```diff
@@ -0,0 +1,381 @@
+      }
+
+      log.debug(`  Found ${events.length} events, total unique PKHs: ${allPKHs.size}`);
+    } catch (err) {
+      log.warn(`Failed to query blocks ${fromBlock}-${toBlock}: ${err.message}`);
+    }
+  }
+
+  log.success(`Found ${allPKHs.size} unique wallet PKHs`);
+  return Array.from(allPKHs);
+}
+
+/**
+ * Get wallet state from Bridge contract
+ */
+async function getWalletState(bridge, pkh) {
+  try {
+    const wallet = await bridge.wallets(pkh);
+    // State enum: Unknown=0, Live=1, MovingFunds=2, Closing=3, Closed=4, Terminated=5
+    const stateNames = ['Unknown', 'Live', 'MovingFunds', 'Closing', 'Closed', 'Terminated'];
+    return {
+      state: stateNames[wallet.state] || 'Unknown',
+      createdAt: parseInt(wallet.createdAt.toString())
+    };
+  } catch (err) {
+    return { state: 'Unknown', createdAt: 0 };
+  }
+}
+
+/**
+ * Main execution
+ */
+async function main() {
+  log.info('='.repeat(60));
+  log.info('tBTC Proof of Funds Query Tool');
+  log.info('='.repeat(60));
+
+  // Connect to Ethereum
+  log.info(`Connecting to Ethereum: ${CONFIG.ethereumRpcUrl.substring(0, 50)}...`);
+  const provider = new ethers.JsonRpcProvider(CONFIG.ethereumRpcUrl);
+
+  // Verify connection
+  const network = await provider.getNetwork();
+  const currentBlock = await provider.getBlockNumber();
+  log.success(`Connected to ${network.name} (chainId: ${network.chainId}), block: ${currentBlock}`);
+
+  // Bridge ABI
+  const bridgeAbi = [
+    'event NewWalletRegistered(bytes32 indexed ecdsaWalletID, bytes20 indexed walletPubKeyHash)',
+    'function wallets(bytes20 walletPubKeyHash) view returns (tuple(bytes32 ecdsaWalletID, bytes32 mainUtxoHash, uint64 pendingRedemptionsValue, uint32 createdAt, uint32 movingFundsRequestedAt, uint32 closingStartedAt, uint32 pendingMovedFundsSweepRequestsCount, uint8 state, bytes32 movingFundsTargetWalletsCommitmentHash))'
+  ];
+
+  const bridge = new ethers.Contract(CONFIG.bridgeAddress, bridgeAbi, provider);
+
+  // Get all wallet PKHs
+  const allPKHs = await getAllWalletPKHs(bridge, provider);
+
+  log.info(`\nQuerying Bitcoin balances for ${allPKHs.length} wallets...`);
+  log.info('(This will take a few minutes due to rate limiting)\n');
+
+  const wallets = [];
+  let totalBalance = 0n;
+
+  for (let i = 0; i < allPKHs.length; i++) {
+    const pkh = allPKHs[i];
+    const cleanPKH = pkh.replace(/^0x/i, '').toLowerCase();
+    const btcAddress = pubKeyHashToBech32Address(cleanPKH);
+
+    log.info(`[${i + 1}/${allPKHs.length}] ${btcAddress.substring(0, 20)}...`);
+
+    // Get wallet state from Ethereum
+    const walletInfo = await getWalletState(bridge, pkh);
+
+    // Get Bitcoin balance
+    const balanceSats = await fetchBitcoinBalance(btcAddress);
+    const balanceBTC = balanceSats / 100000000;
+
+    totalBalance += BigInt(balanceSats);
+
+    // Convert satoshis to BTC as integer string (truncated, no decimals)
+    const balanceBTCInt = Math.floor(balanceSats / 100000000);
```

Comment:
I'm not sure about whether or not this will lead to some significant loss in precision
```suggestion
    const balanceBTCInt = (balanceSats / 100000000).toFixed(8);
```